afqinsight/nn/pt_models.py

```diff
@@ -0,0 +1,166 @@
+import math
+from dipy.utils.optpkg import optional_package
+from dipy.utils.tripwire import TripWire
+
+torch_msg = (
+    "To use afqinsight's pytorch models, you need to have pytorch "
+    "installed. You can do this by installing afqinsight with `pip install "
+    "afqinsight[tf]`, or by separately installing pytorch with `pip install "
+    "pytorch`."
+)
+
+torch, has_torch, _ = optional_package("torch", trip_msg=torch_msg)
+if has_torch:
+    import torch.nn as nn
+else:
+    # Since all model building functions start with Input, we make Input the
+    # tripwire instance for cases where pytorch is not installed.
+    Input = TripWire(torch_msg)
+    print("test")
+
+class mlp4_torch(nn.Module):
+    def __init__(self, input_shape, n_classes, output_activation = torch.softmax, verbose=False):
+        super(mlp4_torch, self).__init__()
+        self.flatten = nn.flatten()
+        self.dropout_1 = nn.Dropout(0.1)
+        self.dropout_2 = nn.Dropout(0.2)
+        self.dropout_3 = nn.Dropout(0.3)
+
+        self.l0 = nn.Linear(input_shape, 500)
+        self.l1 = nn.ReLU(500,500)
+        self.l2 = nn.ReLU(500,500)
+        self.l3 = nn.ReLU(500,500)
```

Comment:
I think there needs to be a `Linear` before every `ReLu` to create the fully connected model.

Reply:
I think you probably want to use `nn.Sequential` to chain together layers into models.